### PR TITLE
fix(functions): manually drop state in function eval_aggr function

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

- Bug Fix
- 
## Related Issues

Continue  #5056

### CODE REVIEW

**Comment by BohuTANG** [comment]
@mergify update

**Review by BohuTANG — APPROVED** [review]

File: common/functions/src/aggregates/aggregator_common.rs [hunk 1]
```diff
@@ -89,16 +89,30 @@ pub fn eval_aggr(
...
+        let mut builder = data_type.create_mutable(1024);
+        func.merge_result(addr, builder.as_mut())?;
+
+        Ok(builder.to_column())
+    };
 
-    Ok(builder.to_column())
+    let result = apply();
+
+    if f.need_manual_drop_state() {
```
Comment (zhang2014) [inline common/functions/src/aggregates/aggregator_common.rs:111]: need `exit_guard`

Comment (sundy-li) [reply common/functions/src/aggregates/aggregator_common.rs:111]: `apply` function is used to catch the whole errors.

`exit_guard ` can not catch the `Bump`.

Comment (zhang2014) [reply common/functions/src/aggregates/aggregator_common.rs:111]: when panic?

Comment (sundy-li) [reply common/functions/src/aggregates/aggregator_common.rs:111]: I think `exit_guard`  can't catch the panic either.

Comment (zhang2014) [reply common/functions/src/aggregates/aggregator_common.rs:111]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a80480b9211961b420a2ce55f0f58cf2

**Pull request merged**